### PR TITLE
adds amendments to the rollout feature flag. Rollout indicative quote…

### PR DESCRIPTION
**Pull request by PirosB3**
Adds some new logic to the `isPriceAwareRFQEnabled` feature flag.

By default, indicative quotes should go through the new price-aware route (regardless of the value in `isPriceAwareRFQEnabled`). Firm quotes will go through the new price-aware route only when enabled by `isPriceAwareRFQEnabled` flag.

### CODE REVIEW

**Review by BMillman19 — APPROVED** [review]

File: packages/asset-swapper/src/utils/market_operation_utils/index.ts [hunk 1]
```diff
@@ -726,8 +728,8 @@ export class MarketOperationUtils {
                         optimizerOpts,
                     );
                 }
-            } else {
-                // A firm quote is being requested. Ensure that `intentOnFilling` is enabled.
+            } else if (!rfqt.isIndicative && isFirmPriceAwareEnabled) {
```
Comment (alexkroeger) [inline packages/asset-swapper/src/utils/market_operation_utils/index.ts:731]: should there be an 'else if...' if priceAwareEnabled is false for both firm and indicative?

Comment (PirosB3) [reply packages/asset-swapper/src/utils/market_operation_utils/index.ts:731]: good question! if firm and indicative quotes are both disabled, none of that if/else block is triggered. In this scenario, the `optimizedOrders` variable is never replaced and proceeds with what was previously set before. This is expected.

Does that make sense? or am I missing smth?

Comment (alexkroeger) [reply packages/asset-swapper/src/utils/market_operation_utils/index.ts:731]: We have routes for

(rfqt.isIndicative && isIndicativePriceAwareEnabled)
and
(!rfqt.isIndicative && isFirmPriceAwareEnabled)

but was wondering if we needed routes for:

(rfqt.isIndicative && !isIndicativePriceAwareEnabled)
and
(!rfqt.isIndicative && !isFirmPriceAwareEnabled)

Or are we only using RFQT if it's price aware?

Comment (PirosB3) [reply packages/asset-swapper/src/utils/market_operation_utils/index.ts:731]: oh, that's a really good question. So, this pathway that starts [here](https://github.com/0xProject/protocol/pull/30/files#diff-863cb8194bd0f4e5a026d8ba07c4a5256dacd6e342a427d6953fa759b28f8239R683) is only accessed when price-aware RFQ is enabled. When Price-aware RFQ is disabled, we still do RFQ but at a different section of the code: [here](https://github.com/0xProject/protocol/pull/30/files#diff-24bb4e154d234a6991fb599588e16823703a26243956e07cacd9ea7722de7ba4L703) for firm quotes and [here](https://github.com/0xProject/protocol/pull/30/files#diff-863cb8194bd0f4e5a026d8ba07c4a5256dacd6e342a427d6953fa759b28f8239R218) for indicative quotes

Comment (PirosB3) [reply packages/asset-swapper/src/utils/market_operation_utils/index.ts:731]: oh, that's a really good question. So, this pathway that starts [here](https://github.com/0xProject/protocol/pull/30/files#diff-863cb8194bd0f4e5a026d8ba07c4a5256dacd6e342a427d6953fa759b28f8239R683) is only accessed when price-aware RFQ is enabled. When Price-aware RFQ is disabled, we still do RFQ but at a different section of the code: [here](https://github.com/0xProject/protocol/pull/30/files#diff-24bb4e154d234a6991fb599588e16823703a26243956e07cacd9ea7722de7ba4L703) for firm quotes and [here](https://github.com/0xProject/protocol/pull/30/files#diff-863cb8194bd0f4e5a026d8ba07c4a5256dacd6e342a427d6953fa759b28f8239R218) for indicative quotes

File: packages/asset-swapper/src/utils/utils.ts [hunk 2]
```diff
@@ -4,35 +4,25 @@ import { BigNumber, NULL_BYTES } from '@0x/utils';
...
-    } else {
-        return {
-            isIndicativePriceAwareEnabled: true,
-            isFirmPriceAwareEnabled: false,
-        };
-    }
+export function getPriceAwareRFQRolloutFlags(priceAwareRFQFlags?: PriceAwareRFQFlags): PriceAwareRFQFlags {
+    return priceAwareRFQFlags !== undefined
+        ? priceAwareRFQFlags
+        : {
```
Comment (BMillman19) [inline packages/asset-swapper/src/utils/utils.ts:22]: Only comment is this default is in two places in the code, here and above in `constants.ts`

**Review by BMillman19 — APPROVED** [review]

File: packages/asset-swapper/src/swap_quoter.ts [hunk 3]
```diff
@@ -700,7 +701,7 @@ export class SwapQuoter {
 
         if (
             opts.rfqt && // This is an RFQT-enabled API request
-            !opts.rfqt.isPriceAwareRFQEnabled && // If Price-aware RFQ is enabled, firm quotes are requested later on in the process.
+            !getPriceAwareRFQRolloutFlags(opts.rfqt.priceAwareRFQFlag).isFirmPriceAwareEnabled && // If Price-aware RFQ is enabled, firm quotes are requested later on in the process.
```
Comment (alexkroeger) [inline packages/asset-swapper/src/swap_quoter.ts:704]: Should this include a similar check for the indicative flag?

e.g !getPriceAwareRFQRolloutFlags(opts.rfqt.priceAwareRFQFlag).isIndicativePriceAwareEnabled &&

Comment (PirosB3) [reply packages/asset-swapper/src/swap_quoter.ts:704]: no, because the firm and indicative quotes are 2 independent flags.

Comment (alexkroeger) [reply packages/asset-swapper/src/swap_quoter.ts:704]: indicative quotes are pulled here in the case of a false `isIndicativePriceAwareEnabled` flag: https://github.com/0xProject/protocol/blob/bf3035ad258f9a02581f7b5fca8ee8c6af047f37/packages/asset-swapper/src/utils/market_operation_utils/index.ts#L369

LGTM

**Review by alexkroeger — APPROVED** [review]

**Pull request merged**